test/gpu_reference_kernel.cpp

```diff
@@ -310,6 +312,29 @@ static std::string miopen_type_to_string(miopenDataType_t type)
     return "n/a";
 }
 
+// input: a vector of lengths of dims in a tensor
+// multiply each element with a random constant integer
+void pad_tensor_strides(std::vector<int>& strides)
+{
+    auto pvec = [](const char* name, const auto& vec) {
+        std::cout << name << ": [";
+        for(const auto& v : vec)
+        {
+            std::cout << v << ", ";
+        }
+        std::cout << "]\n";
+    };
+
+    pvec("orig strides", strides);
+    auto c = prng::gen_A_to_B(1, 3);
+    // int c = 2;
+    for(auto& v : strides)
+    {
+        v = v * c;
```

Comment:
@amberhassaan HIP tidy issue:
```
[2023-09-18T16:25:31.036Z] /var/jenkins/workspace/Open_amber_tests-non-packed-conv/test/gpu_reference_kernel.cpp:333:11: warning: Consider using std::transform algorithm instead of a raw loop. [useStlAlgorithm]
[2023-09-18T16:25:31.036Z]         v = v * c;
[2023-09-18T16:25:31.036Z]           ^
```

Reply:
Thats exactly I was thinking about but decided to keep it!
Didn't know that clang-tidy does that kind of analysis.

Reply:
Tbh, I take issue with this suggestion from `hip-tidy`. The original code is more readable and `std::transform` will obscure it. I would make such a change for a large vector input but that will never happen here. So, I'll silence `hip-tidy` with a comment.

Reply:
@amberhassaan okay but please suppress it only for this snippet of code instead of globally. :)

Reply:
Yes, that was always my intention. This is a small silly loop that might itself change by the end of this PR.